contributingGuides/STYLE.md

```diff
@@ -684,125 +574,9 @@ When writing a stateless component you must ALWAYS add a `displayName` property
-
-`React.PureComponent` has a default implementation of `shouldComponentUpdate` that does a shallow comparison of props and state to determine if it should re-render or not.
-
-Read the [React Docs](https://reactjs.org/docs/react-api.html#reactpurecomponent) to understand this more.
-
-
-```js
-// Internal react code for pure component looks like this with regards to shouldComponentUpdate
-
-if (type.prototype && type.prototype.isPureReactComponent) {
-    shouldUpdate = !shallowEqual(oldProps, props) ||
-                   !shallowEqual(oldState, state);
-}
-```
-
-> A common pitfall when converting from Component to PureComponent is to forget that the children need to re-render too. As with all React - if the parent doesn’t re-render the children won’t either. So if you have a PureComponent with children, those children can only update if the parent’s state or props are shallowly different (causing the parent to re-render). You can only have a PureComponent parent if you know none of the children should re-render if the parent doesn’t re-render.
-
-**Tip:** If you think you need `React.PureComponent`, but you're not using a class component use [`React.memo()`](https://reactjs.org/docs/react-api.html#reactmemo) to achieve the same thing.
-
-*_3. Class based components: Use it when you need to maintain state and use lifecycle methods._*
-
-Always extend from `React.Component` and use a class component when:
-
-- A component needs some data passed to it from a parent holding the data in `this.state`. A class component would hold this data in state and pass it down to the child via props. - If you need to perform some kind of side-effect after the component mounts (`componentDidMount()`) or tweak a component's rendering performance.
-- A final case where you might need to use a class component is if you need to use a `ref`. We have not yet adopted built-in React hooks like `useRef()` so if you need a `ref` use a class component.
-
-```javascript
-// Bad
-const MyComponent = props => {
-    const inputRef = useRef();
-    return <Input ref={inputRef} />;
-};
-
-// Good
-class MyComponent extends React.Component {
-    render() {
-        return <Input ref={el => this.inputRef = el} />;
-    }
-}
-```
-
-*_4. Render Props: Use for Cross-Cutting Concerns and Code Reuse_*
-
-Let's say you have two separate components that render two different things, but you want them both to be based on the same state. You can pass a `render` property to a component or pass a function as a child to accomplish this. You can read more about how this works and when to use it in the [React Docs](https://reactjs.org/docs/render-props.html).
-
-## Do not use inheritance for other React components
```

Comment:
There isn't any concern here since we won't have any class components. Hooks are designed to facilitate code reuse so while there may still be HOCs render props are going to be a less attractive pattern than hooks.